.github/workflows/mirror-docker-image.yml

```diff
@@ -0,0 +1,37 @@
+name: mirror-docker-image
+
+on:
+  workflow_dispatch:
+    inputs:
+      image:
+        type: string
+        description: |
+          oras-compatible image ref to mirror (ex. 'docker.io/library/busybox')
+      version:
+        type: string
+        description: |
+          Version of the source image to mirror (ex. '@<digest>', ':<tag>')
+      output-image:
+        description: |
+          Image path under 'wasmcloud/mirrors' to use (ex. 'a' -> 'wasmcloud/mirrors/a:<tag>', 'tools/b' -> 'wasmcloud/mirrors/tools/b:<tag>')
+        optional: true
+
+permissions: {}
+
+jobs:
+  mirror-docker-image:
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+    steps:
+      - uses: oras-project/setup-oras@ca28077386065e263c03428f4ae0c09024817c93
+      - name: Docker login
+        shell: bash
+        run: |
+          echo ${{ secrets.GITHUB_TOKEN }} | oras login ghcr.io --username ${{ github.repository_owner }} --password-stdin
```

Comment:
We should have a step to also run `oras logout` at the end of the workflow since this isn't a pre-built action that handles post-action clean up.